### PR TITLE
Internal 350 graplctl aws test

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/350
https://github.com/grapl-security/issue-tracker/issues/415
https://github.com/grapl-security/issue-tracker/issues/345
https://github.com/grapl-security/issue-tracker/issues/421
https://github.com/grapl-security/issue-tracker/issues/546

### What changes does this PR make to Grapl? Why?

This PR changes how we do provisioning and end-to-end testing in local grapl such that we can do it _almost_ the same locally as we do in prod.

First, we now provision local grapl with the `provisioner` lambda's code, _almost_ just like in prod.

Additionally, the end-to-end tests are now also executed in a lambda, which allows us to run them _almost_ the same way in AWS as we do locally.

This is enabled with the addition of a `graplctl aws test` command.

### How were these changes tested?

`make test-integration`
`make test-e2e`
Also tested in AWS like this: 
```
make graplctl
make pulumi-prep
pulumi up
graplctl dgraph create
graplctl aws provision
graplctl upload analyzer --analyzer_main_py ./etc/local_grapl/suspicious_svchost/main.py
graplctl upload analyzer --analyzer_main_py ./etc/local_grapl/unique_cmd_parent/main.py
graplctl upload sysmon --logfile ./etc/sample_data/eventlog.xml
graplctl aws test
graplctl dgraph destroy
pulumi destroy
```

### CODE REVIEW

File: test/docker-compose.e2e-tests.yml [hunk 1]
```diff
@@ -5,46 +5,73 @@ version: "3.8"
...
+              upload analyzer \
+              --analyzer_main_py ./etc/local_grapl/unique_cmd_parent/main.py && \
+          graplctl \
+              --grapl-region $AWS_REGION \
+              --grapl-deployment-name $DEPLOYMENT_NAME \
+              --grapl-version $DEPLOYMENT_NAME \
+              --aws-profile default \
+              upload sysmon \
+              --logfile ./etc/sample_data/eventlog.xml && \
+          python -c 'import lambdex_handler; lambdex_handler.handler(None, None)'
```
Comment (christophermaier) [inline test/docker-compose.e2e-tests.yml:35]: I assume this is due to the unzipping of the lambda package. However, why are we doing this if `graplctl` can invoke the e2e test lambda?

File: test/docker-compose.e2e-tests.yml [hunk 2]
```diff
@@ -5,46 +5,73 @@ version: "3.8"
...
+      LAMBDA_ACCESS_KEY_SECRET: ${LAMBDA_ACCESS_KEY_SECRET}
+      LAMBDA_ENDPOINT: ${LAMBDA_ENDPOINT}
+      ROUTE53_ACCESS_KEY_ID: ${ROUTE53_ACCESS_KEY_ID}
+      ROUTE53_ACCESS_KEY_SECRET: ${ROUTE53_ACCESS_KEY_SECRET}
+      ROUTE53_ENDPOINT: ${ROUTE53_ENDPOINT}
+      VSC_DEBUGGER_PORT: ${VSC_DEBUGGER_PORT_FOR_GRAPL_E2E_TESTS}
+      MG_ALPHAS: ${MG_ALPHAS}
+      GRAPL_API_HOST: ${GRAPL_API_HOST}
+      GRAPL_HTTP_FRONTEND_PORT: ${GRAPL_HTTP_FRONTEND_PORT}
+      GRAPL_TEST_USER_NAME: ${GRAPL_TEST_USER_NAME}
```
Comment (christophermaier) [inline test/docker-compose.e2e-tests.yml:74]: Environment variables defined with a map (as opposed to a list) can still inherit their values from the environment by leaving the value empty.

Why did you need to change them all from array to map, though?

https://docs.docker.com/compose/compose-file/compose-file-v3/#environment

**Comment by wimax-grapl** [comment]
also, linking #894

File: src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py [hunk 3]
```diff
@@ -1,35 +1,60 @@
...
+    )["SecretString"]
+
+
+def _sha_and_pepper(username: str, password: str) -> str:
+    # see src/js/engagement_view/src/components/login/utils/passwordHashing.tsx
+    pepper = "f1dafbdcab924862a198deaa5b6bae29aef7f2a442f841da975f1c515529d254"
+    hashed = hashlib.sha256((password + pepper + username).encode("utf-8"))
+    for _ in range(5000):
+        hashed = hashlib.sha256(hashed.hexdigest().encode("utf-8"))
+    return hashed.hexdigest()
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py:35]: No idea, just trying to replicate what's done in `passwordHashing.tsx`

File: test/docker-compose.e2e-tests.yml (hunk 1 above)
Comment (jgrillo-grapl) [reply test/docker-compose.e2e-tests.yml:35]: `graplctl` cannot invoke the test lambda in localstack, unfortunately. It fails and no logs are emitted. I was able to get the `provisioner` lambda to run in localstack, but not the `e2e-test-runner`.

File: build-support/editor_setup.py [hunk 4]
```diff
@@ -42,6 +42,8 @@ class PyrightConfig(TypedDict):
     "exclude": [
         "src/js/**",
         "src/rust/**",
+        "bin/**",
+        "dist/**",
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply build-support/editor_setup.py:46]: pyright was very slow, was trying to figure out why. I thought it might be scanning the zipfiles in these directories so I added them to the ignore list. Will remove.

File: src/python/Dockerfile [hunk 5]
```diff
@@ -236,15 +236,14 @@ CMD source venv/bin/activate && \
...
-#
-# Local provision
-################################################################################
+# test
+FROM grapl-notebook AS grapl-notebook-test
 
-# build
-FROM grapl-python-build AS grapl-provision
+# This file will improve once #444 lands
+# primarily to use pre-downloaded tools instead of downloading again
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/Dockerfile:241]: I'm not sure, I don't recall making this change..

File: test/docker-compose.e2e-tests.yml (hunk 2 above)
Comment (jgrillo-grapl) [reply test/docker-compose.e2e-tests.yml:74]: I believe at some point during the development of these changes I actually needed a mapping, but I can change it back now

File: pulumi/infra/secret.py [hunk 6]
```diff
@@ -42,6 +47,71 @@ def __init__(self, opts: Optional[pulumi.ResourceOptions] = None) -> None:
...
+                            },
+                        ],
+                    }
+                )
+            ),
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+
+
+class TestUserPassword(pulumi.ComponentResource):
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply pulumi/infra/secret.py:80]: I tried doing this with a base class `_Secret(pulumi.ComponentResource)` but it seemed to confuse pulumi.

File: pulumi/infra/secret.py [hunk 7]
```diff
@@ -70,3 +140,11 @@ def grant_read_permissions_to(self, role: aws.iam.Role) -> None:
...
             opts=pulumi.ResourceOptions(parent=role),
         )
+
+        # TODO: What do we do about rotation?
+        self.version = aws.secretsmanager.SecretVersion(
+            "secret",
+            secret_id=self.secret.id,
+            secret_string=self.random_uuid.result,
+            opts=pulumi.ResourceOptions(parent=self),
+        )
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply pulumi/infra/secret.py:150]: more rebase errors

File: src/python/Dockerfile [hunk 8]
```diff
@@ -236,15 +236,14 @@ CMD source venv/bin/activate && \
...
 
-COPY --chown=grapl python/grapl_provision grapl_local_provision
-COPY --chown=grapl --from=grapl-analyzerlib-build /home/grapl/venv venv
+RUN source venv/bin/activate && \
+    pip install nbqa mypy boto3-stubs[essential]
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/Dockerfile:244]: well, the `grapl_provision` stuff is all gone now, but I'm not sure why we're installing these dependencies with pip instead of copying in the virutalenv like before. Like I said below, I don't believe I actually made this change... not sure why it's in this change set.

File: src/python/e2e-test-runner/e2e-test-runner.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+#
+# e2e-tests
+################################################################################
+FROM grapl/graplctl:$TAG AS e2e-tests
+USER root
+RUN apt-get update && apt-get install -y --no-install-recommends unzip
+USER grapl
+COPY --chown=grapl ./dist/src.python.e2e-test-runner.src/lambda.zip lambda.zip
+RUN unzip lambda.zip
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/e2e-test-runner/e2e-test-runner.Dockerfile:11]: We are unzipping the lambda because we can't actually invoke it as a lambda in localstack, instead we have to invoke it "by hand" in this container. Will fix the misnamed package.

File: src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py [hunk 10]
```diff
@@ -17,20 +24,29 @@ class GraphQLException(Exception):
...
     ) -> Dict[str, Any]:
         resp = requests.post(
-            f"{self.endpoint}/graphQlEndpoint/graphql",
+            f"{self.endpoint}/graphql",
             params={"query": query, "variables": json.dumps(variables or {})},
             cookies={"grapl_jwt": self.jwt},
         )
-        assert resp.status_code == HTTPStatus.OK, resp.json()
+        if resp.status_code != HTTPStatus.OK:
+            resp_str = "\\n".join(resp.iter_lines(decode_unicode=True))
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py:43]: Because I don't want the actual log lines to have newlines in them (e.g. when the AssertionError on L44 gets logged). It's easy enough to un-escape these upon logs analysis, it's a far more difficult problem--intractable in general--to deal with multi-line logs.

File: src/js/grapl-cdk/lib/grapl-cdk-stack.ts [hunk 11]
```diff
@@ -33,6 +33,7 @@ import { NodeIdentifier } from "./services/node_identifier";
 import { Provisioner } from "./services/provisioner";
 import { SysmonGraphGenerator } from "./services/sysmon_graph_generator";
 import { OSQueryGraphGenerator } from "./services/osquery_graph_generator";
+import { E2eTestRunner } from "./services/e2e-test-runner";
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/lib/grapl-cdk-stack.ts:36]: I had to make these changes in order to test the e2e-test-runner in AWS, however now that we can deploy grapl with Pulumi in AWS I think I can remove this. If we're using Pulumi moving forward, shouldn't we just remove the `src/js/grapl-cdk` directory completely?

File: src/js/grapl-cdk/package-lock.json [hunk 12]
```diff
@@ -1,11864 +1,8 @@
 {
   "name": "grapl-cdk",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/js/grapl-cdk/package-lock.json:4]: Yeah not sure what happened here. Will try to revert this.

File: src/python/graplctl/graplctl.Dockerfile [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    --gecos '' \
+    --home /home/grapl \
+    --shell /bin/bash \
+    grapl
+USER grapl
+ENV USER=grapl
+WORKDIR /home/grapl
+RUN mkdir -p /home/grapl/bin
+RUN mkdir -p /home/grapl/.aws
+RUN echo "[default]\naws_access_key_id=${AWS_ACCESS_KEY_ID}\naws_secret_access_key=${AWS_SECRET_ACCESS_KEY}" > /home/grapl/.aws/credentials
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/graplctl/graplctl.Dockerfile:17]: The important thing here is to have a `/home/grapl/.aws/credentials` with the `[default]` profile and credentials defined in it. The actual credential values don't seem to matter as we're running this against localstack. If that file doesn't exist and contain the `[default]` profile, we get this error trying to run `graplctl` against localstack:

```
2021-06-03T14:39:34.000668854Z botocore.exceptions.ProfileNotFound: The config profile (default) could not be found
```

[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/graplctl/graplctl.Dockerfile]: I've updated this to include obviously fake credentials in the container. I think this is fine, as this container is only used in the test environment. If you'd prefer I mount a file instead I can figure out how to do that.

File: src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py [hunk 14]
```diff
@@ -26,9 +27,11 @@ def __init__(self, endpoint: str) -> None:
...
     @staticmethod
     def from_env() -> ModelPluginDeployerClient:
-        hostname = os.environ["GRAPL_MODEL_PLUGIN_DEPLOYER_HOST"]
-        port = os.environ["GRAPL_MODEL_PLUGIN_DEPLOYER_PORT"]
-        return ModelPluginDeployerClient(endpoint=f"http://{hostname}:{port}")
+        port = int(os.environ["GRAPL_HTTP_FRONTEND_PORT"]) if IS_LOCAL else 443
+        protocol = "http" if IS_LOCAL else "https"
+        return ModelPluginDeployerClient(
+            endpoint=f'{protocol}://{os.environ["GRAPL_API_HOST"]}:{port}/modelPluginDeployer'
+        )
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py]: done

File: pulumi/infra/secret.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply pulumi/infra/secret.py:80]: https://github.com/grapl-security/issue-tracker/issues/573

File: src/js/grapl-cdk/lib/services/e2e-test-runner.ts [hunk 15]
```diff
@@ -0,0 +1,70 @@
+import * as cdk from "@aws-cdk/core";
+import * as iam from "@aws-cdk/aws-iam";
+import * as lambda from "@aws-cdk/aws-lambda";
+import * as apigateway from "@aws-cdk/aws-apigateway";
+
+import { GraplServiceProps } from "../grapl-cdk-stack";
+import { SchemaDb } from "../schemadb";
+import { Provisioner } from "./provisioner";
```
Comment (wimax-grapl) [inline src/js/grapl-cdk/lib/services/e2e-test-runner.ts:8]: i'm skipping reviewing the CDK code closely

File: src/python/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/python/Dockerfile:241]: Hah, this sucks, it's from
https://github.com/grapl-security/grapl/issues/444
not issue-tracker

File: src/python/Dockerfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/python/Dockerfile:244]: If you didn't make these changes, do you mind if I return them back to how they look in `main`?

File: src/python/e2e-test-runner/e2e_test_runner/test_main.py [hunk 16]
```diff
@@ -52,13 +52,13 @@ def scope_has_N_items() -> bool:
...
 
-        wait_for_one(WaitForCondition(scope_has_N_items), timeout_secs=240)
+        wait_for_one(WaitForCondition(scope_has_N_items), timeout_secs=300)
         gql_client = GraphqlEndpointClient(jwt=EngagementEdgeClient().get_jwt())
         wait_for_one(
             WaitForNoException(
                 lambda: ensure_graphql_lens_scope_no_errors(gql_client, LENS_NAME)
             ),
-            timeout_secs=40,
+            timeout_secs=300,
```
Comment (wimax-grapl) [inline src/python/e2e-test-runner/e2e_test_runner/test_main.py:61]: (would be nice to benchmark these)

File: src/python/grapl-common/grapl_common/env_helpers.py [hunk 17]
```diff
@@ -130,6 +160,8 @@ def _localstack_client(
     return client_create_fn(
         service,
         endpoint_url=f"http://{os.environ['LOCALSTACK_HOSTNAME']}:{os.environ['EDGE_PORT']}",
+        region_name=region,
+        config=config,
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/python/grapl-common/grapl_common/env_helpers.py:151]: ^ was this resolved?

File: src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py [hunk 18]
```diff
@@ -17,20 +19,23 @@ class GraphQLException(Exception):
...
+            raise AssertionError(
```
Comment (wimax-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py:35]: feels like the wrong exception class, but also it's test code so Whatever

File: src/python/graplctl/graplctl.Dockerfile [hunk 19]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl.Dockerfile:1]: not a must-have, but a nice-to-have-but-feel-free-to-ignore:

could you add a comment on why a dockerized graplctl is needed? (I *believe* it's so e2e can invoke stuff with it)

File: src/python/graplctl/graplctl/aws/commands.py [hunk 20]
```diff
@@ -32,101 +21,25 @@ def aws(
 
 
 @aws.command()
-@click.option(
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl/aws/commands.py:35]: feels good man

File: src/python/graplctl/graplctl/aws/lib.py [hunk 21]
```diff
@@ -18,108 +16,32 @@
...
+    if status == 200 and result.get("FunctionError") is None:
         for line in logs.splitlines():
             LOGGER.info(line)
-        LOGGER.info("provisioner lambda succeeded")
+        LOGGER.info(f"lambda invocation succeeded for {function_name}")
     else:
-        for line in logs.splitlines():
-            LOGGER.error(line)
+        LOGGER.error(
+            f"{''.join(l.decode('utf-8') for l in cast(StreamingBody, result['Payload']).iter_lines())}"
```
Comment (wimax-grapl) [inline src/python/graplctl/graplctl/aws/lib.py:35]: 1. could we, like, bring this out to a separate function or something
2. looks like the f-string is no longer required

File: src/python/provisioner/provisioner/app.py [hunk 22]
```diff
@@ -108,15 +112,20 @@ def _retrieve_test_user_password(
 
 
 def provision(event: Any = None, context: Any = None) -> None:
+    LOGGER.info("provisioning grapl")
+
     graph_client = GraphClient()
-    dynamodb: DynamoDBServiceResource = boto3.resource("dynamodb")
-    secretsmanager: SecretsmanagerClient = boto3.client("secretsmanager")
+    dynamodb: DynamoDBServiceResource = DynamoDBResourceFactory(boto3).from_env()
+    secretsmanager: SecretsmanagerClient = SecretsManagerClientFactory(boto3).from_env()
```
Comment (wimax-grapl) [inline src/python/provisioner/provisioner/app.py:119]: fwiw, since those factories are well typed, the typedecl is not needed

File: test/docker-compose.e2e-tests.yml [hunk 23]
```diff
@@ -5,16 +5,34 @@ version: "3.8"
...
+              --aws-profile default \
+              upload analyzer \
+              --analyzer_main_py ./etc/local_grapl/unique_cmd_parent/main.py && \
+          graplctl \
+              --grapl-region $AWS_REGION \
+              --grapl-deployment-name $DEPLOYMENT_NAME \
+              --grapl-version $DEPLOYMENT_NAME \
+              --aws-profile default \
+              upload sysmon \
+              --logfile ./etc/sample_data/eventlog.xml && \
```
Comment (wimax-grapl) [inline test/docker-compose.e2e-tests.yml:34]: I don't love how repetitive this is, but... whatever

File: test/docker-compose.e2e-tests.yml [hunk 24]
```diff
@@ -5,16 +5,34 @@ version: "3.8"
...
+          python -c 'import lambdex_handler; lambdex_handler.handler(None, None)'
```
Comment (wimax-grapl) [inline test/docker-compose.e2e-tests.yml:35]: i feel like i'm staring into the abyss

Comment (jgrillo-grapl) [reply test/docker-compose.e2e-tests.yml:35]: lol yeah pretty much 😬

File: src/python/grapl-common/grapl_common/env_helpers.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/grapl-common/grapl_common/env_helpers.py:151]: No, we can probably remove region here? The issue I ran into is effectively the fact that graplctl's idea of a "grapl region" isn't the same thing as AWS_REGION and/or AWS_DEFAULT_REGION. So I'm not sure what the "right" way to reconcile all this is, but we can certainly remove this `region` param and fix the consequences. Injecting the `config` here is important because we use it in various contexts to tweak timeouts.

File: src/python/graplctl/graplctl.Dockerfile (hunk 19 above)
Comment (jgrillo-grapl) [reply src/python/graplctl/graplctl.Dockerfile:1]: Yeah, we need a container with graplctl installed in it so we can execute the provisioner lambda in localstack and upload the various analyzers and test datasets while running the e2e tests locally.

Comment (jgrillo-grapl) [reply src/python/graplctl/graplctl.Dockerfile:1]: This effectively serves as an integration test of those `graplctl` commands. Unfortunately we can't make it a "complete" integration test because for some unknown reason the e2e-test-runner lambda won't run in localstack (but the provisioner will).

**Review by wimax-grapl — APPROVED** [review]

File: src/python/Dockerfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/Dockerfile:244]: hmm actually, this does look necessary because `grapl_provision` is no longer a thing. I guess I did make this change? I don't remember it but who knows. I don't understand what the comment on L240-241 means regarding "#444" so I'm really not sure what to do here. I propose we leave it for now and investigate in a follow-up story, because I'd really rather not go down the rabbit hole of reverse-engineering this now: 
https://github.com/grapl-security/issue-tracker/issues/574

File: src/python/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/Dockerfile:241]: See also https://github.com/grapl-security/issue-tracker/issues/574

File: src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py (hunk 18 above)
Comment (jgrillo-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/graphql_endpoint_client.py:35]: The `assert` call this code replaces (to correct the issue with `resp.json()` not working as intended) raises `AssertionError` when it fails.

File: src/python/grapl-common/grapl_common/env_helpers.py [hunk 25]
```diff
@@ -220,9 +400,11 @@ class SecretsManagerClientFactory(FromEnv["SecretsManagerClient"]):
     def __init__(self, boto3_module: Any):
         self.client_create_fn = boto3_module.client
 
-    def from_env(self) -> SecretsManagerClient:
+    def from_env(
+        self, region: Optional[str] = None, config: Optional[Config] = None
+    ) -> SecretsManagerClient:
         client: SecretsManagerClient = _client_get(
-            self.client_create_fn, _SecretsManagerParams
+            self.client_create_fn, _SecretsManagerParams, region=region, config=config
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/python/grapl-common/grapl_common/env_helpers.py]: Should I create a story to refactor this? I think it's important to make sure we can inject the config, but I don't really have any opinions beyond that.

File: src/python/grapl-common/tests/test_env_helpers.py [hunk 26]
```diff
@@ -0,0 +1,306 @@
...
+            assert config.region_name == "preferred-fake-region"
+            result = _client_get(
+                client_create_fn=lambda _, region_name, config: region_name,
+                params=_CLIENT_GET_PARAMS,
+                config=config,
+            )
+            assert result == "preferred-fake-region"
+
+    def test_client_get_returns_expected_when_config_present_and_region_present_and_AWS_DEFAULT_REGION_and_AWS_REGION_set(
+        self,
```
Comment (wimax-grapl) [inline src/python/grapl-common/tests/test_env_helpers.py:281]: this might be the longest test function name i've ever seen lol (it's fine)

File: src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py [hunk 27]
```diff
@@ -1,35 +1,64 @@
...
+from grapl_common.env_helpers import SecretsManagerClientFactory
+from grapl_common.grapl_logger import get_module_grapl_logger
+from grapl_tests_common.clients.common import endpoint_url
 
 _JSON_CONTENT_TYPE_HEADERS = {"Content-type": "application/json"}
 _ORIGIN = {
     "Origin": "https://local-grapl-engagement-ux-bucket.s3.amazonaws.com",
 }
 
+LOGGER = get_module_grapl_logger(default_log_level="DEBUG")
```
Comment (wimax-grapl) [inline src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py:17]: not a blocker, but it's sort of odd for the default to be DEBUG

**Review by wimax-grapl — APPROVED** [review]

File: src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py (hunk 27 above)
Comment (jgrillo-grapl) [reply src/python/grapl-tests-common/grapl_tests_common/clients/engagement_edge_client.py:17]: I think it makes sense in test code, regardless I believe we're passing `GRAPL_LOG_LEVEL=DEBUG` into the environment (both in lambda and in the docker containers). If it becomes a problem can change it.
